lib/ansible/executor/module_common.py

```diff
@@ -1186,9 +1186,15 @@ def _find_module_utils(module_name, b_module_data, module_path, module_args, tas
                     # so that no one looking for the file reads a partially
                     # written file)
                     if not os.path.exists(lookup_path):
-                        # Note -- if we have a global function to setup, that would
-                        # be a better place to run this
-                        os.makedirs(lookup_path)
+                        try:
+                            # Note -- if we have a global function to setup, that would
+                            # be a better place to run this
+                            os.makedirs(lookup_path)
```

Comment:
in [py3](https://docs.python.org/3/library/os.html#os.makedirs) we can use `os.makedirs(lookup_path, exists_ok=True)`. With that we can get rid of the try/except and check condition, condensing everything here into a single line.

Reply:
I'm assuming we're only concerned with py3 given the use of `FileExistsError`; that didn't exist in py2. If we need py2 support we'd need something like so to trap the exception in a py2 compatible way and the above comment wont work; the `exists_ok` flag was new in py3.

``` python
import errno

class FileExistsError(OSError):
    def __init__(self, message):
        super(FileExistsError, self).__init__(errno.EEXIST, message)
```

Reply:
@cloudnull see makedirs_safe for older python compatibility (probably will be in backport), but since we require python >=3.8 on controller, `exist_ok=True` should be good enough

Reply:
For changes in `devel`, we are only concerned with supporting Python >3.8. If we decide to backport an equivalent fix, we could use `makedirs_safe()`.

Reply:
Since we're intending to backport this fix, it'll have to remain a `try/except OSError:` with a double-check in the exception handler. Once it's landed in devel and our CI is able to handle py3-only controller tests (which is still WIP), we can update to use py3-only bits.

Reply:
@nitzmahone no need, makedirs_safe already does that

Reply:
It involves `unfrackpath` among other things ... :fearful:

Reply:
That was my hesitancy as well. I think I'll go back to the plain vanilla `try/except`.